aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -184,9 +185,7 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
 
       if (persistFailure instanceof AwsServiceException) {
         int statusCode = ((AwsServiceException) persistFailure).statusCode();
-        if (statusCode >= 500 && statusCode < 600) {
-          commitStatus = CommitStatus.FAILURE;
-        } else {
+        if (!(statusCode >= 500 && statusCode < 600)) {
```

Comment:
For me, this is harder to read, because of the additional negation.

Reply:
Thanks, @Fokko for your review. I have reverted this change now.